### PR TITLE
Use github actions for our pull request validations

### PR DESCRIPTION
**Pull request by tapthaker**
Changes:

- Adds `github-actions-pull-request.yml` and removes Travis-specific scripts. 
- Use `opam exec -- ` for executing `opam` actions so that the correct environment ( `PATH`, etc) is available
- Sleep for 1 sec before executing build for `providers` because of the race condition because it depends on the `NEAL` package.

### CODE REVIEW

File: src/Makefile.propagate [hunk 1]
```diff
@@ -15,6 +15,7 @@ test: $(NEAL) $(patsubst %/Makefile,%.test,$(SUBDIRS))
 	@true
 
 %.build:
+	[ "$*" != "providers" ] || sleep 1
```
Comment (ghvg1313) [inline src/Makefile.propagate:18]: this is sleeping on `make providers`?
can you explain a bit more on what's the race condition?

**Comment by ghvg1313** [comment]
How come Travis CI is still running, thought it's getting pulled?

File: src/Makefile.propagate (hunk 1 above)
Comment (tapthaker) [reply src/Makefile.propagate:18]: The `src/core` module gets compiled and is then installed as a local package `neal` for opam ( OCaml Package Manager ) inside the `register` task here: https://github.com/uber/NEAL/blob/master/src/Makefile. 

However, the `src/providers` which depend on this local `neal` package starts to build as soon as the build task for `neal` is complete and  thus complains about `Could not find module: neal`

**Comment by tapthaker** [comment]
> How come Travis CI is still running, thought it's getting pulled?

Need to pull our Travis from our Github Apps. Looks like this is for someone who has admin authorisation for uber github org

File: src/Makefile.propagate (hunk 1 above)
Comment (ghvg1313) [reply src/Makefile.propagate:18]: sounds more like a dependency chain setup issue
ok with this for now, would rather fix the makefile if possible
(is 1s good enough?)

**Review by ghvg1313 — APPROVED** [review]

File: src/Makefile.propagate (hunk 1 above)
Comment (tapthaker) [reply src/Makefile.propagate:18]: Ya, "RECURSIVE MAKE CONSIDERED HARMFUL" https://accu.org/journals/overload/14/71/miller_2004/

**Review by ivoreroman — APPROVED** [review]

**Pull request merged**